src/main/java/com/exactprosystems/remotehand/windows/actions/TableSearch.java

```diff
@@ -0,0 +1,144 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactprosystems.remotehand.windows.actions;
+
+import com.exactprosystems.remotehand.RhUtils;
+import com.exactprosystems.remotehand.ScriptExecuteException;
+import com.exactprosystems.remotehand.windows.ElementSearcher;
+import com.exactprosystems.remotehand.windows.WindowsAction;
+import com.exactprosystems.remotehand.windows.WindowsDriverWrapper;
+import com.exactprosystems.remotehand.windows.WindowsSessionContext;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+
+public class TableSearch extends WindowsAction
+{
+	private static final Logger loggerInstance = LoggerFactory.getLogger(TableSearch.class);
+	private static final String FILTER = "filter";
+	private static final String TARGET_COLUMN = "column";
+	private static final String INDEX = "index";
+	private static final String FIRST_ROW_INDEX = "firstrowindex";
+	private static final String DEFAULT_FIRST_ROW_NUMBER = "0";
+	private static final String ROW_NAME_FORMAT = "rowNameFormat";
+	private static final String DEFAULT_ROW_NAME_FORMAT = "Row %s";
+	private static final String ROW_ELEMENT_NAME_FORMAT = "rowelementnameformat";
+	private static final String DEFAULT_ROW_ELEMENT_NAME_FORMAT = "%s row %s";
+	private static final String ROW_ELEMENT_VALUE_FORMAT = "rowelementvalueformat";
+	private static final String DEFAULT_ELEMENT_VALUE_FORMAT = "Value.Value";
+
+	private String rowNameFormat;
+	private String rowElementNameFormat;
+	private String rowElementValueFormat;
+	private String targetColumnName;
+	private int columnIndex;
+	private int firstRowIndex;
+	private Map<String, String> filtersMap;
+
+	@Override
+	public String run(WindowsDriverWrapper driverWrapper, Map<String, String> params,
+			WindowsSessionContext.CachedWebElements cachedElements) throws ScriptExecuteException
+	{
+		try {
+			handleInputParams(params);
+			ElementSearcher elementSearcher = new ElementSearcher(params, driverWrapper.getDriver(), cachedElements);
+			WebElement table = elementSearcher.searchElement();
+			setTimeOut(driverWrapper, 0);
+
+			WebElement row;
+			boolean rowFound = false;
+			int i = firstRowIndex;
+			while (!rowFound) {
```

Comment:
Deprecated TableClick action allows us to use different indexes with RowName and child ValueName.
So there are Row started with 1, but values started with 0
But there is no possible to do this.